sklearn/metrics/cluster/unsupervised.py

```diff
@@ -56,6 +65,18 @@ def silhouette_score(X, labels, metric='euclidean', sample_size=None,
         <sklearn.metrics.pairwise.pairwise_distances>`. If X is the distance
         array itself, use ``metric="precomputed"``.
 
+    block_size : int, optional, default=64
```

Comment:
Okay then can we have it as `memory_usage` and let it denote the total memory consumed for all the jobs. And add a note in the docstring stating that this is not the maximum memory as there could be some more overheads?

Because as a user, I'd blindly set this to my ram maximum. And `n_jobs` to number of processors I have...